chanfitness/rate_limit.go

```diff
@@ -0,0 +1,40 @@
+package chanfitness
+
+import "time"
+
+var (
+	// rateLimitScale is the number of events we allow per rate limited
+	// tier. Increasing this value makes our rate limiting more lenient,
+	// decreasing it makes us less lenient.
+	rateLimitScale = 200
+
+	// rateLimits is the set of rate limit tiers we apply to our peers
+	// based on their flap count. A peer can be placed in their tier by
+	// dividing their flap count by the rateLimitScale and returning the
+	// value at that index.
+	rateLimits = []time.Duration{
+		time.Second,
+		time.Second * 5,
+		time.Second * 30,
+		time.Minute,
+		time.Minute * 30,
+		time.Hour,
+	}
+)
+
+// getRateLimit returns the value of the rate limited tier that we are on based
+// on current flap count. If a peer's flap count exceeds the top tier, we just
+// return our highest tier.
+func getRateLimit(flapCount uint32) time.Duration {
```

Comment:
does `flapcount`being `uint32` add much? I'd just use an `int` to avoid all these casts. (Afaik Golang uses 64 bit ints in 64 bit archs, but I could be wrong)